modules/whoisquery/collect.go

```diff
@@ -0,0 +1,27 @@
+package whoisquery
+
+import (
+	"fmt"
+)
+
+func (x *WhoisQuery) collect() (map[string]int64, error) {
+	remainingTime, err := x.prov.remainingTime()
+	if err != nil {
+		return nil, err
+	}
+
+	if remainingTime == -1 {
+		return nil, fmt.Errorf("no domain was provided by '%s'", x.Config.Source)
+	}
```

Comment:
i guess `if err != nil` below is enough, isnt it? 

is it possible to get nil error and still not actual value?